cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -118,6 +118,24 @@ impl InstallableLocker for Pkgdb {
         debug!("locking installable: {pkgdb_cmd:?}");
 
         let lock = call_pkgdb(pkgdb_cmd).map_err(|err| match err {
+            CallPkgDbError::PkgDbError(PkgDbError {
+                exit_code: error_codes::NIX_GENERIC,
+                context_message:
+                    Some(ContextMsgError {
+                        caught: Some(nix_error),
+                        ..
+                    }),
+                ..
+            }) => FlakeInstallableError::LockInstallable(nix_error.message),
+            CallPkgDbError::PkgDbError(PkgDbError {
+                exit_code: error_codes::NIX_EVAL,
+                context_message:
+                    Some(ContextMsgError {
+                        caught: Some(nix_error),
+                        ..
+                    }),
+                ..
+            }) => FlakeInstallableError::LockInstallable(nix_error.message),
```

Comment:
nonblocking: should our catch all case print the caught message? Or are we pretty confident we've got everything now?

Reply:
I'm not confident that we've got everything just because I don't know all of the ways that Nix can fail to evaluate a package. I will say that both the `hello-unfree` case and the `cachix` case were caught by just the `NIX_GENERIC` branch I added, and I included the `NIX_EVAL` case just to be safe.

If by "catch all case" you mean
```
_ => FlakeInstallableError::Pkgdb(err),
```
then I think that's a lot more effort than it's worth since that's falling back to the existing `PkgDbError` machinery (which I'll admit is pretty hard to reason about).

Reply:
I was wondering if we should have
```
            CallPkgDbError::PkgDbError(PkgDbError {
                context_message:
                    Some(ContextMsgError {
                        caught: Some(caught_error),
                        ..
                    }),
                ..
            }) => FlakeInstallableError::LockInstallable(caught_error.message),
```
rather than picking out individual exit codes. But I've lost track of which errors we want the caught message so I'm not sure

Reply:
Yeah it's hard to say. The `caught` message in general could be something like "file not found" which doesn't make sense without the `context` message that says something like "failed to open the lockfile".

I don't know that we have any cases here that follow that exact pattern though.